jbmc/src/java_bytecode/java_object_factory.cpp

```diff
@@ -161,6 +163,14 @@ class java_object_factoryt
     const exprt &max_length_expr,
     size_t depth,
     update_in_placet update_in_place,
+    const source_locationt &location,
+    const array_element_generatort &element_generator);
+
+  code_blockt assign_element(
+    const exprt &element_at_counter,
```

Comment:
I don't immediately understand "element_at_counter", will see if I have a useful suggestion when I find the implementation...